include/deal.II/fe/mapping_q_generic.h

```diff
@@ -81,8 +81,54 @@ template <int,int> class MappingQ;
  * a manifold object to all cells that can then also be used by the mapping
  * classes for higher order mappings.
  *
+ * <h4>Behavior along curved boundaries and with different manifolds</h4>
```

Comment:
The question should be whether we want to put the full description in `MappingQ` or in `MappingQGeneric`. I would actually prefer to put it in `MappingQGeneric` because that is the more "sane" mapping in my opinion.